src/App.tsx

```diff
@@ -15,47 +16,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+const getReorderedGoods = (
+  goods: string[], { sortType, isReversed }: ReorderOptions,
+) => {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  // Sort and reverse goods if needed
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  visibleGoods.sort((good1, good2) => {
+    if (sortType === SortType.ALPHABET) {
+      return good1.localeCompare(good2);
+    }
+
+    if (sortType === SortType.LENGTH) {
+      return good1.length - good2.length;
+    }
+
+    return 0;
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+};
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+  const goods = getReorderedGoods(
+    [...goodsFromServer], { sortType, isReversed },
+  );
+
+  const sort = (text: SortType) => {
+    setSortType(text);
+  };
+
+  const reverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const reset = (text: SortType) => {
+    setSortType(text);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          // className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortType !== SortType.ALPHABET,
+          })}
+          onClick={() => sort(SortType.ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortType !== SortType.LENGTH,
+          })}
+          onClick={() => sort(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': isReversed !== true,
+          })}
+          onClick={reverse}
         >
           Reverse
         </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortType !== SortType.NONE || isReversed)
+        && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => reset(SortType.NONE)}
```

Comment:
There is no need to pass an argument here `onClick={() => reset(SortType.NONE)}`, since we are setting default values in the function, like with `setIsReversed(false)`
const reset = (text: SortType) => {
    setSortType(text); <------------------SortType.NONE
    setIsReversed(false);
  };